pandas/io/sas/sas_xport.py

```diff
@@ -265,10 +265,6 @@ def __init__(
         else:
             # Copy to BytesIO, and ensure no encoding
             contents = filepath_or_buffer.read()
-            try:
-                contents = contents.encode(self._encoding)
-            except UnicodeEncodeError:
-                pass
             self.filepath_or_buffer = BytesIO(contents)
```

Comment:
Is it possible to determine if it is a binary file handle already, so that one can avoid copying the file to a BytesIO and the having to reread this?

Reply:
I think the conversion to BytesIO is always unnecessary on Python 3. Removed this conversion and added a comment with my thoughts.

Reply:
Per https://www.loc.gov/preservation/digital/formats/fdd/fdd000464.shtml, rows are padded with Null bytes, so I think reading an XPORT file in text-mode on Python 3 will always fail.